frontend/model/contracts/group.js

```diff
@@ -194,17 +194,11 @@ function memberLeaves ({ memberID, dateLeft, heightLeft, ourselvesLeaving }, { c
   // to be rotated. Later, this will be used by 'gi.contracts/group/rotateKeys'
   // (to actually perform the rotation) and Chelonia (to unset the flag if
   // they are rotated by somebody else)
-  // TODO: Improve this API. Developers should not modify state that is managed
-  // by Chelonia.
-  // Example: sbp('chelonia/contract/markKeyForRevocation', contractID, 'csk')
-  if (!state._volatile) state['_volatile'] = Object.create(null)
-  if (!state._volatile.pendingKeyRevocations) state._volatile['pendingKeyRevocations'] = Object.create(null)
-
-  const CSKid = findKeyIdByName(state, 'csk')
-  const CEKid = findKeyIdByName(state, 'cek')
-
-  state._volatile.pendingKeyRevocations[CSKid] = true
-  state._volatile.pendingKeyRevocations[CEKid] = true
+  sbp('chelonia/queueInvocation', contractID, () => {
+    return sbp('chelonia/contract/setPendingKeyRevocation', contractID, ['cek', 'csk'])
```

Comment:
The other versions of this are accompanied by a call to `'gi.actions/out/rotateKeys'`.. where does that happen for this, and if it doesn't happen, why not?

Reply:
That's done later on `gi.contracts/group/leaveGroup`